DSharpPlus.UnifiedCommands/Internals/CommandModuleRegister.cs

```diff
@@ -0,0 +1,344 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DSharpPlus.UnifiedCommands.Internals;
+
+internal static class CommandModuleRegister
+{
+    internal static void RegisterMessageCommands(MessageFactory factory, Assembly assembly)
+    {
+        IEnumerable<Type> classes = assembly.GetTypes()
+            .Where(t => t.IsClass &&
+                        !t.IsAbstract &&
+                        t.GetCustomAttribute<MessageModuleAttribute>() is not null &&
+                        t.IsPublic &&
+                        t.IsSubclassOf(typeof(MessageModule)));
+
+        NullabilityInfoContext nullabilityContext = new();
+
+        foreach (Type? @class in classes)
+        {
+            MessageModuleData module = new()
+            {
+                Factory = ActivatorUtilities.CreateFactory(@class, Array.Empty<Type>())
+            };
+
+            MethodInfo[] methods = @class.GetMethods();
+            if (methods.Length == 0)
+            {
+                continue; // Not worth saving this class if it is has no methods.
+            }
+
+            foreach (MethodInfo method in methods)
+            {
+                MessageAttribute? attribute = method.GetCustomAttribute<MessageAttribute>();
+                if (attribute is null || !method.IsPublic)
+                {
+                    continue;
+                }
+
+                string[] moduleName = @class.GetCustomAttribute<MessageModuleAttribute>()?.Name?.Split(' ') ??
+                                      Array.Empty<string>();
+                string[] methodName = attribute.Name.Split(' ');
+
+                List<MessageParameterData> parameters = new();
+                bool cantConsumeMoreArguments = false;
+                foreach (ParameterInfo parameter in method.GetParameters())
+                {
+                    MessageOptionAttribute? paramAttribute = parameter.GetCustomAttribute<MessageOptionAttribute>();
+                    MessageParameterData parameterData = new();
+
+                    if (paramAttribute is null)
+                    {
+                        if (cantConsumeMoreArguments)
+                        {
+                            throw new InvalidMessageModuleStructure(
+                                $"You cannot have more arguments for method {method.Name}.");
+                        }
+
+                        parameterData.IsPositionalArgument = true;
+                        parameterData.Name = parameter.Name ?? string.Empty;
+
+                        if (
+                            parameter.GetCustomAttribute<RemainingArgumentsAttribute>() is not null)
+                        {
+                            if (parameter.ParameterType != typeof(string))
+                            {
+                                throw new InvalidMessageModuleStructure(
+                                    "You need to use string if you are using RemainingTextAttribute.");
```

Comment:
Would be nice if the error message contains the methodname, so the user can see easily where they have to change something